strs.go

```diff
@@ -358,6 +364,38 @@ func (db *RoseDB) incrDecrBy(key []byte, incr int64) (int64, error) {
 	return valInt64, nil
 }
 
+func (db *RoseDB) incrDecrByFloat64(key []byte, incr float64) (float64, error) {
+	val, err := db.getVal(db.strIndex.idxTree, key, String)
+	if err != nil && !errors.Is(err, ErrKeyNotFound) {
+		return 0, err
+	}
+	if bytes.Equal(val, nil) {
+		val = []byte("0")
+	}
+	valFloat64, err := strconv.ParseFloat(string(val), 64)
+	if err != nil {
+		return 0, ErrWrongValueType
+	}
+
+	if (incr < 0 && valFloat64 < 0 && incr < (math.MinInt64-valFloat64)) ||
+		(incr > 0 && valFloat64 > 0 && incr > (math.MaxInt64-valFloat64)) {
```

Comment:
I thought `valFloat64 < 0` in first case and `valFloat64 > 0` in second case is useless. Maybe this **if statement** could be   considered again?

Reply:
Why ？ I still don't understand.

Reply:
because if `valFloat64` is a negative and `incr` is negative too, it will checked by `incr < (math.MinInt64 - valFloat64)` , and if `valFloat64` is not negative , `incr < (math.MinInt64 - valFloat64)` will never be true, so `valFloat64 < 0` is not necessary. Second case is same to what i said upon.

Reply:
if `valFloat64` is a negative and `incr` is negative too, if will be checked by `incr < (math.minInt64 - valFloat64)`,
if `valFloat64` is not negative, it will be checked `incr < (math.MaxInt64 - valFloat64)`, is will not always be true

Reply:
oh I see, that's my fault, thank you for my a lot of questions.